lib/cartopy/mpl/geoaxes.py

```diff
@@ -234,14 +234,25 @@ def __init__(self, axes, **kwargs):
         super().__init__(self._original_path, **kwargs)
         self._axes = axes
 
+        # We need to use a TransformWrapper as our transform so that we can
+        # update the transform without breaking others' references to this one.
+        self._trans_wrap = mtransforms.TransformWrapper(self.get_transform())
+
+    def set_transform(self, transform):
+        self._trans_wrap.set(transform)
+        super().set_transform(self._trans_wrap)
+
     def set_boundary(self, path, transform):
         self._original_path = path
         self.set_transform(transform)
         self.stale = True
 
     def _adjust_location(self):
         if self.stale:
-            self._path = self._original_path.clip_to_bbox(self.axes.viewLim)
+            self.set_path(self._original_path.clip_to_bbox(self.axes.viewLim))
+            # Clip path uses assume path is not changing so need to trigger an
+            # update.
+            self.get_transform().invalidate()
```

Comment:
Just do:
```suggestion
            self._trans_wrap.invalidate()
```
It should propagate up without having to fetch the full transform.